src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2130,6 +2134,30 @@ protected override int GetOrCreateSlot(Symbol symbol, int containingSlot = 0, bo
                 }
             }
 
+            // Share a slot between backing field and associated property/event in the context of a constructor which owns initialization of that backing field.
+            if (this._symbol is MethodSymbol constructor
+                && constructor.IsConstructor()
+                && constructor.IsStatic == symbol.IsStatic)
```

Comment:
>constructor.IsStatic == symbol.IsStatic [](http://example.com/codeflow?start=19&length=39)

Are we testing this case? That is, are we testing use of static property in an instance constructor?

Reply:
This check is only *really* going to be relevant in a scenario like the following:
1. static property is used in instance constructor, *and*
2. somehow, backing field of the same static property is also used in instance constructor. I think this is not possible.

I am adding some tests to poke at that area, but, even if the tests don't show the necessity of the check, I think I will feel more comfortable having it there.